tests/acceptance/test_async.py

```diff
@@ -108,47 +108,51 @@ def example_task():
     assert len(jobs) == 1
 
 
-async def test_abort(async_app: app_module.App):
-    @async_app.task(queue="default", name="task1", pass_context=True)
-    async def task1(context):
-        while True:
-            await asyncio.sleep(0.02)
-            if await context.should_abort_async():
-                raise JobAborted
-
-    @async_app.task(queue="default", name="task2", pass_context=True)
-    def task2(context):
-        while True:
-            time.sleep(0.02)
-            if context.should_abort():
-                raise JobAborted
+@pytest.mark.parametrize("listen_notify", [False, True])
+@pytest.mark.parametrize("sync_task", [False, True])
+async def test_abort(async_app: app_module.App, listen_notify, sync_task):
+    if sync_task:
```

Comment:
I'm not a huge fan of `if` statements in the tests. When what varies cannot be expressed as simple values in the parametrize, I prefer duplicating the test.

(I like the test code to read as simply as possible. Procrastinate tests need to be somewhat smart because it's hard to test complex things simply, but let's not add to that by trying to be too DRY. Being DRY is usually nice for normal code, but it's much less of a requirement for test code)